pkg/operator/baremetal_pod.go

```diff
@@ -330,6 +331,10 @@ func createContainerMetal3IronicApi(config *OperatorConfig) corev1.Container {
 			setEnvVar("HTTP_PORT", "http_port"),
 			setEnvVar("PROVISIONING_INTERFACE", "provisioning_interface"),
 		},
+		LivenessProbe: &corev1.Probe{
+			Handler:             corev1.Handler{HTTPGet: &corev1.HTTPGetAction{Port: intstr.FromInt(6385)}},
```

Comment:
That's the port where Apache listens, isn't it? I'm not sure why we make that configurable. 

Ironic does always listen on 6385, AFAICT.

Do we need to be specifying a path here, too? Maybe something that actually hits ironic's database, so we know it's all the way up? Or is it good enough for us to know that it's responding at all, because we care when it doesn't?

Reply:
I had hard-coded this because I saw the port hard-coded somewhere else, but now I'm not sure where that was. It might have been in the developer deployment files in baremetal-operator, which at the time I didn't realize are not relevant for this.

I was going to make this use the healthcheck middleware, but it's not enabled in our existing config. I'm not sure it would accomplish a lot more than this anyway. I wasn't sure if I could call anything else without authentication. This was sufficient for handling the case where the node went down unexpectedly, but I'm open to suggestions on a better path to call.

Reply:
If we just care about being able to communicate with the service at all, this is probably fine. If we end up needing to be able to tell if the API is up but the DB is down, we can set up the health check for that or pick a different endpoint. None of it is authenticated, so we could list nodes or something, but that's going to get expensive if we don't need it.

Reply:
Ah, I wasn't crazy. We hardcode the port in the ironic-image repo: https://github.com/metal3-io/ironic-image/blob/7bffafd8fb5c141047b64ee2128a0ffe8701d3c2/configure-ironic.sh#L26

I'm inclined to go with this as-is and if we find that we need more detailed health checking in the future we can always add it.

Reply:
I concur.